File: Factorial/fact.fs
(* fsharpi --quiet --exec fact.fs *)

module fact 

(*
let rec factorial n =
    if n <= 0 then 1
    else n * factorial (n - 1);;
*)

let rec factorial = function
    | n when n<=0 -> 1
    | n -> n * factorial (n-1);;

let main = printfn "%d" (factorial 4);;

File: Factorial/fact.ml
(* ocaml fact.ml *)

(*
let rec fact n =
    if (n<=0) then 1 else n * fact (n-1);;
*)

let rec factorial n = 
    match n with
        | n when (n>0) -> n * factorial (n-1)
        | _ -> 1;;

let ans = Printf.sprintf "%d" (factorial 4);;
print_endline ans;;
